### PR TITLE
2.20.4-build

### PR DESCRIPTION
**Pull request by ahmadiesa-abu**
Added change_boot_order operation , and new ISO node type

### CODE REVIEW

File: cloudify_vsphere/devices/__init__.py [hunk 1]
```diff
@@ -564,3 +564,100 @@ def detach_pci_device(ctx, **kwargs):
         else:
             raise NonRecoverableError(
                 'PCI Device can\'t be detached while VM is running')
+
+
+def _get_device_keys(vm, device_type):
+    device_keys = []
+    for device in vm.config.hardware.device:
+        if isinstance(device, device_type):
+            # if 'Hard disk 1' in device.deviceInfo.label:
```
Comment (EarthmanT) [inline cloudify_vsphere/devices/__init__.py:573]: Why commented?

File: cloudify_vsphere/devices/__init__.py [hunk 2]
```diff
@@ -564,3 +564,100 @@ def detach_pci_device(ctx, **kwargs):
...
+            "device_type": vim.vm.device.VirtualEthernetCard,
+            "boot_obj": vim.vm.BootOptions.BootableEthernetDevice,
+            "keys_required": True
+        }
+    }
+    vsphere_server_id = ctx.instance.runtime_properties.get(
+        'vsphere_server_id')
+    connection_config_props = ctx.node.properties.get(
+        'connection_config')
+    cl = ServerClient()
```
Comment (EarthmanT) [inline cloudify_vsphere/devices/__init__.py:621]: Don't we have a decorator for getting a client?

File: plugin.yaml [hunk 3]
```diff
@@ -720,6 +720,37 @@ node_types:
...
+        type: cloudify.datatypes.vsphere.Config
+    interfaces:
+      cloudify.interfaces.lifecycle:
+        create:
+          implementation: vsphere.vsphere_storage_plugin.cidata.upload_iso
+          inputs: {}
+        delete:
+          implementation: vsphere.vsphere_storage_plugin.cidata.delete
+          inputs: {}
+
```
Comment (EarthmanT) [inline plugin.yaml:753]: Did you add these deprecations to cfy-lint?

File: vsphere_storage_plugin/cidata.py [hunk 4]
```diff
@@ -91,3 +91,32 @@ def delete(rawvolume_client, **kwargs):
...
+@op
+@with_rawvolume_client
+def upload_iso(rawvolume_client,
+               datacenter_name,
+               allowed_datastores,
+               allowed_datastore_ids,
+               volume_prefix,
+               iso_file_path,
+               **_):
+    is_node_deprecated(ctx.node.type)
```
Comment (EarthmanT) [inline vsphere_storage_plugin/cidata.py:105]: We can probably replace with with the deprecation_warning decorator.

File: vsphere_storage_plugin/cidata.py [hunk 5]
```diff
@@ -91,3 +91,32 @@ def delete(rawvolume_client, **kwargs):
...
+    if ctx.instance.runtime_properties.get(VSPHERE_STORAGE_FILE_NAME):
```
Comment (EarthmanT) [inline vsphere_storage_plugin/cidata.py:106]: Is this runtime property cleaned up on uninstall? For example, maybe install failed. They ran rollback. But the property is already set, but we do not actually have the instance.

File: cloudify_vsphere/devices/__init__.py (hunk 1 above)
Comment (ahmadiesa-abu) [reply cloudify_vsphere/devices/__init__.py:573]: will remove, that comment , it is not needed , the devices would be sorted correctly when listed from the vm

File: plugin.yaml (hunk 3 above)
Comment (ahmadiesa-abu) [reply plugin.yaml:753]: will add a PR on cfy-lint for this

File: vsphere_storage_plugin/cidata.py (hunk 4 above)
Comment (ahmadiesa-abu) [reply vsphere_storage_plugin/cidata.py:105]: will create another task to go over all the plugin and change the logic there, maybe make check deprecation part of op decorator

**Pull request merged**